check/src/typecheck.rs

```diff
@@ -1041,11 +1041,8 @@ impl<'a> Typecheck<'a> {
             s.push(c as char);
             let symbol = self.symbols.symbol(&s[..]);
             if self.type_variables.get(&symbol).is_none() {
-                self.type_variables.insert(symbol,
-                                           Type::variable(TypeVariable {
-                                               id: level,
-                                               kind: Kind::typ(),
-                                           }));
+                let var = TypeVariable::new(level, Kind::hole());
+                self.type_variables.insert(symbol, Type::variable(var));
```

Comment:
In practice I don't think it matters but I think it would be more correct if `next_variable` takes the kind as a parameter instead. The method is only called in a single place where it can easily take the kind of the `Type::Variable` it replaced.